src/ctapipe/tools/compute_irf.py

```diff
@@ -0,0 +1,637 @@
+                "Using `fov_offset_bins = [valid_offset.min, valid_offset.max]`."
+            )
+            hdus.append(
+                create_rad_max_hdu(
+                    rad_max=self.opt_result.spatial_selection_table["cut"].reshape(
+                        -1, 1
+                    ),
+                    reco_energy_bins=np.append(
+                        self.opt_result.spatial_selection_table["low"],
+                        self.opt_result.spatial_selection_table["high"][-1],
+                    ),
+                    fov_offset_bins=u.Quantity(
+                        [
+                            self.opt_result.valid_offset.min,
+                            self.opt_result.valid_offset.max,
+                        ]
+                    ).reshape(-1),
+                )
+            )
+        return hdus
+
+    def _make_benchmark_hdus(self, hdus):
+        hdus.append(
+            self.bias_resolution_maker(
+                events=self.signal_events[self.signal_events["selected"]],
+            )
+        )
+        hdus.append(
+            self.angular_resolution_maker(
+                events=self.signal_events[self.signal_events["selected_gh"]],
+            )
+        )
+        if self.do_background:
+            if self.opt_result.spatial_selection_table is None:
+                raise ValueError(
+                    "Calculating the point-source sensitivity requires "
+                    f"theta cuts, but {self.cuts_file} does not contain any."
+                )
+
+            hdus.append(
+                self.sensitivity_maker(
+                    signal_events=self.signal_events[self.signal_events["selected"]],
+                    background_events=self.background_events[
+                        self.background_events["selected_gh"]
+                    ],
+                    spatial_selection_table=self.opt_result.spatial_selection_table,
+                    gamma_spectrum=self.gamma_target_spectrum,
+                )
+            )
+        return hdus
+
+    def start(self):
+        """
+        Load events and calculate the irf (and the benchmarks).
+        """
+        reduced_events = dict()
+        for sel in self.particles:
+            if sel.epp.gammaness_classifier != self.opt_result.clf_prefix:
+                raise RuntimeError(
+                    "G/H cuts are only valid for gammaness scores predicted by "
+                    "the same classifier model. Requested model: %s. "
+                    "Model used for g/h cuts: %s."
+                    % (
+                        sel.epp.gammaness_classifier,
+                        self.opt_result.clf_prefix,
+                    )
+                )
+
+            if (
+                sel.epp.quality_query.quality_criteria
+                != self.opt_result.precuts.quality_criteria
+            ):
+                self.log.warning(
+                    "Precuts are different from precuts used for calculating "
```

Comment:
Why are we calling this now "precuts" when we chose the term "QualityQuery" precisely to avoid the term "precuts"...

Reply:
Good point, fixed it.